File: shopper-web/src/app/model/ShoppingList.ts
import {ListItem} from "./ListItem";
import {CategoryPreset} from "./CategoryPreset";

export class ShoppingList {
    id: number;
    name: string;
    items: ListItem[];
    ownerId: string;
    ownerName: string;
    ownerAvatar: any;
    isOwner: boolean;
    done: number;
    archived: boolean;
    shared: string[];
    preset: CategoryPreset;

    constructor() {

    }
}

File: shopper-watch-companion/app/src/main/java/pl/qprogramming/shopper/watch/model/ShoppingList.java
package pl.qprogramming.shopper.watch.model;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ShoppingList {
    private Long id;
    private String name;
    private String ownerName;
    private boolean archived;
    private CategoryPreset preset;
    private Long done;
    private List<ListItem> items;
}
